app/services/CurrenciesService.js

```diff
@@ -18,27 +18,111 @@ module.exports = BaseService.extends({
   classname: 'CurrenciesService',
 
   getAllRateInfo: function (options, callback) {
+    console.log("************* run to get rate in service")
+    const startTime = new Date().getTime()
+
 
     const db = this._getDbConnection();
     //const tradeAdapter = db.model().getSlaveAdapter();
     const rateAdapter = db.model('Rate7dModel').getSlaveAdapter();
 
     let pairs = {}
 
-    Object.keys(tokens).forEach(token => {
-      if ((token.toUpperCase() !== "ETH") &&
-        !tokens[token].delisted &&
-        helper.shouldShowToken(token)) {
-        pairs[token] = (asyncCallback) => this._getRateInfo(token, {
-          //tradeAdapter: tradeAdapter,
-          rateAdapter: rateAdapter
-        }, asyncCallback)
+    const nowInSeconds = Utils.nowInSeconds();
+    const DAY_IN_SECONDS = 24 * 60 * 60;
+    const dayAgo = nowInSeconds - DAY_IN_SECONDS;
+    const hour30Ago = nowInSeconds - 30 * 60 * 60;
+    const hour18Ago = nowInSeconds - 18 * 60 * 60;
+    const weekAgo = nowInSeconds - DAY_IN_SECONDS * 7;
+
+
+    const lastSql = `
+      SELECT mid_expected as 'rate_24h', quote_symbol
+      FROM rate7d
+      WHERE block_timestamp IN (
+          SELECT MAX(block_timestamp)
+          FROM rate7d
+          where block_timestamp < ?
+      );
+    `;
+    const lastParams = [hour18Ago];
+
+
+    const pointSql = `
+      SELECT AVG(mid_expected) as rate7d, quote_symbol FROM rate 
+      WHERE mid_expected > 0 AND block_timestamp >= ? GROUP BY quote_symbol, h6_seq
```

Comment:
similar to rate_24h, we should inject quote_symbol IN (list of symbol).

Reply:
okey